### PR TITLE
Add location method to search builder

### PR DESCRIPTION
**Pull request by EdwardNelson**
In response to feedback from #77. I have simply made this PR to see what it would feel like to use and what other people think. 

In the other PR I spoke about additional features that Algolia and Elasticsearch provide. When I had a think and stripped them down to the essentials and what is compatible between Algolia and Elasticsearch I found it was only Geosearch that was key.

I don't expect this to be merged, I just want this to start a discussion about whether scout is already supporting the maximum features it plans to.

### CODE REVIEW

File: src/Engines/AlgoliaEngine.php [hunk 1]
```diff
@@ -162,4 +164,19 @@ public function getTotalCount($results)
...
     }
+
+    /**
+     * Merge location data with options array to pass to Algolia.
+     *
+     * @param \Laravel\Scout\Builder $builder
+     * @param array $options
+     * @return array
+     */
+    private function mergeLocation (Builder $builder, array $options)
```
Comment (taylorotwell) [inline src/Engines/AlgoliaEngine.php]: Why is there a space after the function name?

Comment (EdwardNelson) [inline src/Engines/AlgoliaEngine.php]: My bad, fixed

**Comment by EdwardNelson** [comment]
See #12 for more discussion

**Comment by taylorotwell** [comment]
Holding off on this for now.

**Comment by julienbourdeau** [comment]
What about now? It's an interesting feature I think.

**Pull request closed without merging**